src/lib/asn1/oid_maps.cpp

```diff
@@ -16,338 +13,1357 @@ namespace Botan {
 
 namespace {
 
-std::vector<std::pair<const char*, std::vector<uint32_t>>> oid_maps() {
+// The hash can collide so we must verify the actual value matches before returning
+std::optional<std::string> if_match(const OID& oid, std::initializer_list<uint32_t> val, const char* name) {
+   if(oid.matches(val)) {
+      return std::string(name);
+   } else {
+      return {};
+   }
+}
```

Comment:
Technically, this could also return a `std::optional<std::string_view>` given that all the callers pass a static string literal. This would postpone the need of an allocation of `std::string` to the caller of `lookup_static_oid`.
In practice however, this is called via `OID_Map::oid2str()` exclusively which itself currently returns a string. And I'm not sure it is feasible (read: safe) to let this return a string view from there as well.

Nevertheless, I would argue that this generated code should be as efficient as possible. So pushing the `std::string` allocation into `oid2str()` would make sense to me.

Reply:
Done